engine/apps/alerts/models/alert_receive_channel.py

```diff
@@ -519,42 +519,35 @@ def heartbeat_module(self):
     # Demo alerts
     def send_demo_alert(self, force_route_id=None, payload=None):
         logger.info(f"send_demo_alert integration={self.pk} force_route_id={force_route_id}")
+
+        if not self.is_demo_alert_enabled:
+            raise UnableToSendDemoAlert("Unable to send demo alert for this integration.")
+
         if payload is None:
             payload = self.config.example_payload
-        if self.is_demo_alert_enabled:
```

Comment:
I tried to simplify things a bit and reduce number of indentation levels. 
The only logical change is checking for `if not self.is_demo_alert_enabled` before anything.